src/legacy/core_plugins/vis_default_editor/public/components/controls/time_interval.tsx

```diff
@@ -25,6 +25,81 @@ import { FormattedMessage } from '@kbn/i18n/react';
 
 import { search, AggParamOption } from '../../../../../../plugins/data/public';
 import { AggParamEditorProps } from '../agg_param_props';
+const { parseEsInterval, InvalidEsCalendarIntervalError } = search.aggs;
+
+function getInvalidEsMessage(interval: string) {
+  return i18n.translate(
+    'visDefaultEditor.controls.timeInterval.invalidCalendarIntervalErrorMessage',
+    {
+      defaultMessage: 'Invalid calendar interval: {interval}, value must be 1',
+      values: { interval },
+    }
+  );
+}
+
+// we check if Elasticsearch interval is valid to show a user appropriate error message
+// e.g. there is the case when a user inputs '14d' but it's '2w' in expression equivalent and the request will fail
+// we don't check it for 0ms because the overall time range has not yet been set
+function isValidCalendarInterval(interval: string) {
+  if (interval === '0ms') {
+    return true;
+  }
+
+  try {
+    parseEsInterval(interval);
+    return true;
+  } catch (e) {
+    if (e instanceof InvalidEsCalendarIntervalError) {
+      return false;
+    }
+
+    return true;
+  }
+}
+
+const errorMessage = i18n.translate(
+  'visDefaultEditor.controls.timeInterval.invalidFormatErrorMessage',
+  {
+    defaultMessage: 'Invalid interval format.',
+  }
+);
+
+function validateInterval(
+  agg: any,
+  value?: string,
+  definedOption?: ComboBoxOption,
+  timeBase?: string
+) {
+  if (definedOption) {
+    return { isValid: true };
+  }
+
+  if (!value) {
+    return { isValid: false, error: errorMessage };
```

Comment:
Previously, we didn't show an error message, when value was not defined.
Not it shows `Invalid interval format.` , but the value is not specified so I would skip the error message as it was before. IMHO

Reply:
nice catch! thanks, I fixed it.